### PR TITLE
Bump jekyll-generator-single-source

### PR DESCRIPTION
**Pull request by fabianrbz**
Update the site to use the new version of `jekyll-generator-single-source`.

Most notable changes:
* we added support for labeling releases
* URLs now use `/{{ page.release }}/` instead of `/{{ page.version }}` which is more accurate
* the way to handle `dev` release changes slightly, instead of having a `app/_data/docs_nav_dev.yml` it needs to be named after the next release, e.g. `app/_data/docs_nav_2.6.x.yml`. The `dev` bit is handled via the `app/_data/versions.yml` file, by adding `label: dev` to the corresponding version.

This also updates:
* ruby + dependencies
* simplifies the way to run the app locally `make clean && make run` builds and runs the site, including redirects

Did you sign your commit? [Instructions](https://github.com/kumahq/.github/blob/main/CONTRIBUTING.md#sign-your-commits)

Have you read [Contributing guidelines](https://github.com/kumahq/.github/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

**Comment by fabianrbz** [comment]
@lahabana test is failing due to a caching issue, can you clear the cache for that action?

**Comment by lahabana** [comment]
@fabianrbz is this ready to review?

**Comment by fabianrbz** [comment]
@lahabana yes

**Comment by slonka** [comment]
@fabianrbz - tests are failing, can you look into them?

**Comment by lahabana** [comment]
@fabianrbz are you still looking into this?

**Comment by slonka** [comment]
@fabianrbz are you still looking into this?

**Comment by fabianrbz** [comment]
@lahabana @slonka sorry, I'm not working on this atm, I might be able to find some time after summit though

File: app/_data/docs_nav_kuma_2.10.x.yml [hunk 1]
```diff
@@ -1,5 +1,6 @@
 generate: true
 assume_generated: true
+release: 2.10.x
```
Comment (lahabana) [inline app/_data/docs_nav_kuma_2.10.x.yml:3]: NIT: It's unfortunate to have to have this here when it's already in the file name. It means more places to edit when adding a new release and therefore more likeliness to get it wrong

Comment (fabianrbz) [reply app/_data/docs_nav_kuma_2.10.x.yml:3]: We can remove this in the future, it was done to make it backwards compatible with kuma who had `_dev` in the file name.

**Review by lahabana — APPROVED** [review]

**Pull request merged**